tests/de/gurkenlabs/litiengine/util/geom/GeometricUtilitiesTests.java

```diff
@@ -191,6 +202,278 @@ public void testDeltaX(double angle, double expectedDeltaX){
+  })
+  public void testGetPerpendicularIntersection(double x1, double y1, double x2, double y2, double pointX, double pointY){
+    // arrange
+    Point2D point = new Point2D.Double(pointX, pointY);
+    Line2D line = new Line2D.Double(x1, y1, x2, y2);
+    double distanceToLine = line.ptLineDist(point); // shortest distance between the point and the line
+
+    // act
+    Point2D intersection = GeometricUtilities.getPerpendicularIntersection(point, line);
+
+    // assert, if the distance from the line to the point is equal to the distance between the intersection and the
+    //  point, this means that the intersection point is correct (distance measured at closest point, which is true
+    //  if the intersection line is perpendicular)
+    assertEquals(distanceToLine, intersection.distance(point), 0.000001d);
+  }
+
+  @ParameterizedTest(name = "testGetPointOnCircle circleCenter=({0},{1}), circleRadius={2}, angle={3}, expectedPoint=({4},{5})")
+  @CsvSource({
+          "0d, 0d, 1.0d, 0d, 1.0d, 0d", // 0 degrees
+          "0d, 0d, 1.0d, 45d, 0.7071d, 0.7071d", // 45 degrees
+          "0d, 0d, 1.0d, 90d, 0d, 1.0d", // 90 degrees
+          "0d, 0d, 1.0d, 180d, -1.0d, 0d", //  180 degrees
+          "0d, 0d, 1.0d, 270d, 0d, -1.0d", // 270 degrees
+          "0d, 0d, 1.0d, 360d, 1.0d, 0d", // 360 degrees = 0 degrees
+          "1.5d, 2.74d, 2.0d, 84.84d, 1.679874d, 4.73189d" // 84.84 degrees, offset center
+  })
+  public void testGetPointOnCircle(double centerX, double centerY, double radius, double angle, double expectedX, double expectedY){
+    // arrange
+    Point2D center = new Point2D.Double(centerX, centerY);
+
+    // act
+    Point2D pointOnCircle = GeometricUtilities.getPointOnCircle(center, radius, angle);
+
+    // assert
+    assertEquals(expectedX, pointOnCircle.getX(), 0.0001d);
+    assertEquals(expectedY, pointOnCircle.getY(), 0.0001d);
+  }
+
+  @Test
+  public void testGetAveragePosition(){
+    // arrange
+    Collection<Point2D> points = new ArrayList<>();
+    points.add(new Point2D.Double(2.03d, 4.93d));
+    points.add(new Point2D.Double(-1.78d, 3.12d));
+    points.add(new Point2D.Double(1.053d, 0d));
+
+    // act
+    Point2D avgPosition = GeometricUtilities.getAveragePosition(points);
+
+    // assert
+    assertEquals(0.4343d, avgPosition.getX(), 0.0001d);
+    assertEquals(2.6833d, avgPosition.getY(), 0.0001d);
+  }
+
+  @Test
+  public void testGetAveragePositionEmpty(){
+    // arrange
+    Collection<Point2D> points = new ArrayList<>();
+
+    // act
+    Point2D avgPosition = GeometricUtilities.getAveragePosition(points);
+
+    // assert
+    assertNull(avgPosition);
+  }
+
+  /**
+   * TODO: Sometimes, floating point precision can stand in the way of calculating accurate intersections, this
+   *  should probably be investigated. @see{@link java.awt.geom.RectangularShape#intersects(Rectangle2D)}
+   *  The inaccurate cases are marked in @see{getIntersectsEllipseArguments}
+   */
```

Comment:
In this test case, we found an issue. Seems to be the same as in https://github.com/gurkenlabs/litiengine/pull/397#discussion_r625813854